plugins/datadog_tracer/lib/samson_datadog_tracer/apm.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+module SamsonDatadogTracer
+  module APM
+    class << self
+      def included(clazz)
+        clazz.extend ClassMethods
+      end
+
+      def trace_method_execution_scope(scope_name)
+        if SamsonDatadogTracer.enabled?
+          Datadog.tracer.trace("Custom/Hooks/#{scope_name}") do
+            yield
+          end
+        else
+          yield
+        end
+      end
+    end
+
+    module ClassMethods
+      def trace_method(method)
```

Comment:
this looks kinda funky ... can you add a code comment on what is going on here ?

Reply:
done